components/ambient-api-server/test/e2e/rbac_e2e_test.sh

```
@@ -0,0 +1,1183 @@
+# that a zero-binding user cannot create an arbitrary binding.
+# If sessionFactory were nil, this would succeed (no checks).
+api POST "/role_bindings" "$TOKEN_C" "{\"role_id\":\"${ROLE_PLATFORM_ADMIN}\",\"scope\":\"global\",\"user_id\":\"rbac-user-c\"}"
+assert_status "403" "$HTTP_STATUS" "Zero-binding user cannot create platform:admin binding (escalation checks active)"
+
+# ============================================================
+echo ""
+echo -e "${BOLD}Phase 24: Platform Viewer Cannot Escalate to Admin${NC}"
+
+# Grant User C platform:viewer (via direct DB insert since we need a global binding)
+# We can't grant global from a non-admin, so we use the seed-admin pattern via kubectl
+VIEWER_GLOBAL_BIND=""
```

Shellcheck (0.11.0)

[warning] VIEWER_GLOBAL_BIND appears unused. Verify use (or export if used externally).

(SC2034)